File: cpp/ecbenchmark/src/function/Sphere.cpp
#include "ecbenchmark/function/Sphere.h"

namespace ecb {
   /* Source:
    * @incollection{springer-dif.evo,
     Author = {Price, Kenneth and Storn, Rainer M. and Lampinen, Jouni A.},
     title={{Appendix}},
     booktitle = {Differential Evolution: A Practical Approach to Global Optimization},
     series={Natural Computing Series},
     editor={Rozenberg, G. and Bäck, Th. and Eiben, A. E. and Kok, J. N. and Spaink, H. P.},
     publisher = {Springer-Verlag Berlin Heidelberg},
     pages = {513-533},
     year = {2005}
     }
    * -5.12 - 5.12
    */

    scalar Sphere::f(const std::vector<scalar>& x) {
        scalar result = 0.0;
        for (size_t  i = 0; i < x.size(); ++i) {
            result += x[i] * x[i];
        }
        return result;
    }
}

File: java/ecbenchmark/src/ecbenchmark/function/Sphere.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.function;

import ecbenchmark.Function;

/**
 *
 * @author jcrada
 */
public class Sphere implements Function {

    public Sphere() {
    }

    public double f(final double[] x) {
        double result = 0.0;
        for (int i = 0; i < x.length; ++i) {
            result += x[i] * x[i];
        }
        return result;
    }
}
